cmd/catalogd/main.go

```diff
@@ -246,18 +248,40 @@ func run(ctx context.Context) error {
 	cacheOptions := crcache.Options{
 		ByObject: map[client.Object]crcache.ByObject{},
 	}
-	if cfg.globalPullSecretKey != nil {
-		cacheOptions.ByObject[&corev1.Secret{}] = crcache.ByObject{
-			Namespaces: map[string]crcache.Config{
-				cfg.globalPullSecretKey.Namespace: {
-					LabelSelector: k8slabels.Everything(),
-					FieldSelector: fields.SelectorFromSet(map[string]string{
-						"metadata.name": cfg.globalPullSecretKey.Name,
-					}),
-				},
+
+	saKey, err := sautil.GetServiceAccount()
+	if err != nil {
+		setupLog.Error(err, "Unable to get pod namesapce and serviceaccount")
+		return err
+	}
+
+	setupLog.Info("Read token", "serviceaccount", saKey)
```

Comment:
```suggestion
	if saKey.Namespace == "" || saKey.Name == "" {
		setupLog.Error(nil, "Extracted service account name or namespace is empty", "saKey", saKey)
		return fmt.Errorf("invalid service account identity extracted: %v", saKey)
	}

	setupLog.Info("Successfully extracted serviceaccount identity from token", "namespace", saKey.Namespace, "name", saKey.Name)
```

Reply:
There's no point in checking for an empty name or namespace, as the `sautil.GetServiceAccount()` function should (will) error out if both can't be retrieved.